var/spack/repos/builtin/packages/rose/package.py

```diff
@@ -85,17 +93,46 @@ def languages(self):
 
     def configure_args(self):
         spec = self.spec
-        cc = self.compiler.cc
-        cxx = self.compiler.cxx
+
+        if '+intel_backend' in spec:
+            cc = which('mpicc')
+            cxx = which('mpic++')
+        else:
+            cc = self.compiler.cc
+            cxx = self.compiler.cxx
+
+        if spec.satisfies('@0.9.8:'):
+            edg = "4.12"
+        else:
+            edg = "4.9"
+
         return [
             '--disable-boost-version-check',
+            '--enable-edg_version={0}'.format(edg),
             "--with-alternate_backend_C_compiler={0}".format(cc),
             "--with-alternate_backend_Cxx_compiler={0}".format(cxx),
             "--with-boost={0}".format(spec['boost'].prefix),
             "--enable-languages={0}".format(",".join(self.languages)),
             "--with-z3={0}".format(spec['z3'].prefix) if '+z3' in spec else '',
             '--disable-tests-directory' if '+tests' not in spec else '',
             '--enable-tutorial-directory={0}'.format('no'),
+            '--without-java' if '+java' not in spec else '',
+            '--with-CXX_DEBUG=-g' if '+debug' in spec else '',
+            '--with-C_OPTIMIZE=-O0' if '+optimized' in spec else '',
+            '--with-CXX_OPTIMIZE=-O0' if '+optimized' in spec else '',
```

Comment:
Instead of adding a bunch of empty strings to the list of arguments, why not do something like:
```python
args = [
    '--disable-boost-version-check',
    ...
]

if '+java' in spec:
    args.append('--with-java')
else:
    args.append('--without-java')

return args
```